packages/react-internal/src/components/Modal/Modal.types.ts

```diff
@@ -63,6 +63,12 @@ export interface IModalProps extends React.RefAttributes<HTMLDivElement>, IAcces
    */
   componentRef?: IRefObject<IModal>;
 
+  /**
+   * Optional ref to access the modal container. Use this to access the container div element
+   * properties of the modal.
+   */
+  modalContainerRef?: (container: HTMLDivElement | null) => void;
```

Comment:
I'm not sure about either the naming or the type here. 

Including "modal" in the name seems redundant when the whole component is called Modal, so maybe just `containerRef` or `contentRef`? (Or open to other suggestions.)

Also, the type should probably be `React.Ref` (or `RefObject` from utilities) instead of specifically a callback.

Might also be good to update the documentation to make it a little clearer that this is the container of the actual modal content, whereas `ref` is the layer element.

Reply:
> Thanks for the contribution, but I'm not sure this is quite the right solution.
> 
> I have a couple comments below on the current solution, but I wonder if a better solution would be to modify the behavior of the existing `ref` prop so that it's used on the content/container element rather than the root Layer element. That's technically non-standard but I suspect would overall better match the behavior that users expect. I remember discussing this with @dzearing and/or @joschect internally but I don't remember if we came to any conclusions.

I'm not sure I understand the suggestion here. It sounds like you'd want to move where the `componentRef` prop is added but that sounds like an issue for backwards compatibility for people who already rely on that. Right? So why not just add this additional prop for consumers who require more control?

Additionally, we need this to access the HTMLDivElement to access its properties (I explained why in the PR description).

Reply:
> I'm not sure about either the naming or the type here.
> 
> Including "modal" in the name seems redundant when the whole component is called Modal, so maybe just `containerRef` or `contentRef`? (Or open to other suggestions.)
> 
> Also, the type should probably be `React.Ref` (or `RefObject` from utilities) instead of specifically a callback.
> 
> Might also be good to update the documentation to make it a little clearer that this is the container of the actual modal content, whereas `ref` is the layer element.

Happy to rename this and update the documentation. I would like to push back on making it a RefObject because using that didn't seem to give me the benefits that I got from the callback refs where we gain more control over when the refs are set and unset, which we don't get from a standard ref ([link to source](https://reactjs.org/docs/refs-and-the-dom.html)).

Reply:
I'm suggesting moving the `ref` prop, not `componentRef`.

I suggested `React.Ref` or `IRefObject` from `@fluentui/utilities` because either of those can take either a function or an object.